optax/_src/transform.py

```diff
@@ -393,10 +393,12 @@ def init_fn(params):
   def update_fn(updates, state, params=None):
     del params
     mu = _update_moment(updates, state.mu, b1, 1)
-    nu = _update_moment(updates - state.mu, state.nu, b2, 2)
+    prediction_error = jax.tree_multimap(lambda g, m: g-m, updates, state.mu)
+    nu = _update_moment(prediction_error, state.nu, b2, 2)
     count_inc = _safe_int32_increment(state.count)
     mu_hat = _bias_correction(mu, b1, count_inc)
-    nu_hat = _bias_correction(nu + eps, b2, count_inc)
+    nu_nonzero = jax.tree_map(lambda v: v + eps, nu)
```

Comment:
Why are you adding an epsilon here as well?
The eps is already add in line 403.

Adam typically only adds it *after* the bias correction.
Why does adabelief need an additional epsilon term *before* the bias correction?

Reply:
There's this formula in the appendix:
![image](https://user-images.githubusercontent.com/13261989/96906578-5f23b380-1470-11eb-84a6-d649ef758212.png)
But I believe that's just the `eps_root`, I hadn't noticed it existed before, my bad!